src/test/ui/error-codes/E0301.stderr

```diff
@@ -4,6 +4,15 @@ error[E0301]: cannot mutably borrow in a pattern guard
 LL |         option if option.take().is_none() => {},
    |                   ^^^^^^ borrowed mutably in pattern guard
 
-error: aborting due to previous error
+error[E0596]: cannot borrow immutable local variable `option` as mutable
```

Comment:
this diagnostic is hilarious. I'm assuming this is borrowcheck triggering after errors were reported during match checking. I think the "prevent mutable borrows in pattern guards" should be part of borrowck and not match check. Maybe we can treat pattern guards similar to `Fn` closures?

Reply:
Funnily enough we already have a ticket to move this error to borrowck because it's imprecise https://github.com/rust-lang/rust/issues/45600